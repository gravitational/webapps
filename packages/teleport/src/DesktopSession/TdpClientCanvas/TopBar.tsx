/*
Copyright 2021 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
import React from 'react';
import styled, { useTheme } from 'styled-components';
import { Text, TopNav, Flex } from 'design';
import { Clipboard } from 'design/Icon';
import { colors } from 'teleport/Console/colors';
import ActionMenu from '../ActionMenu';

export default function TopBar(props: Props) {
  const { userHost, clipboard, recording, onDisconnect } = props;
  const theme = useTheme();

  const primaryOnTrue = (b: boolean): any => {
    return {
      color: b ? theme.colors.text.primary : theme.colors.text.secondary,
    };
  };

  const userHostStyle = {
    color: theme.colors.text.secondary,
  };

  const clipboardTextStyle = {
    ...primaryOnTrue(clipboard),
    verticalAlign: 'text-bottom',
  };

  const clipboardStyle = {
    ...primaryOnTrue(clipboard),
    fontWeight: theme.fontWeights.bold,
    fontSize: theme.fontSizes[4] + 'px',
    alignSelf: 'center',
  };

  const recordingTextStyle = primaryOnTrue(recording);

  const recordingIndicatorStyle = {
    width: '10px',
    height: '10px',
    borderRadius: '10px',
    marginRight: '6px',
    backgroundColor: recording
      ? theme.colors.error.light
      : theme.colors.text.secondary,
    verticalAlign: 'text-bottom',
  };

  return (
    <TopNav
      height={`${TopBarHeight}px`}
      bg={colors.dark}
      style={{
        justifyContent: 'space-between',
      }}
    >
      <Text px={3} style={userHostStyle}>
        {userHost}
      </Text>

      <Text style={clipboardTextStyle}>
        <Clipboard style={clipboardStyle} pr={2} />
        Clipboard Sharing {clipboard ? 'Enabled' : 'Disabled'}
      </Text>

      <Flex px={3}>
        <Flex alignItems="center">
          <RecordingIndicator style={recordingIndicatorStyle} />
          <Text style={recordingTextStyle}>
            {recording ? '' : 'Not '}Recording
          </Text>
        </Flex>
        <ActionMenu onDisconnect={onDisconnect} />
      </Flex>
    </TopNav>
  );
}

export const TopBarHeight = 40;

const RecordingIndicator = styled.div``;

type Props = {
  userHost: string;
  clipboard: boolean;
  recording: boolean;
  onDisconnect: VoidFunction;
};
