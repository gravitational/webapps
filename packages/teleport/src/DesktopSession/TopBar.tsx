/*
Copyright 2021 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
import React from 'react';
import styled, { useTheme } from 'styled-components';
import { Text, TopNav, Flex, Button } from 'design';
import { Clipboard, FolderShared } from 'design/Icon';

import { colors } from 'teleport/Console/colors';

import ActionMenu from './ActionMenu';

export default function TopBar(props: Props) {
  const {
    userHost,
    clipboardSharingEnabled,
    onDisconnect,
    canShareDirectory,
    isSharingDirectory,
    onShareDirectory,
  } = props;
  const theme = useTheme();

  const primaryOnTrue = (b: boolean): any => {
    return {
      color: b ? theme.colors.text.primary : theme.colors.text.secondary,
    };
  };

  return (
    <TopNav
      height={`${TopBarHeight}px`}
      bg={colors.dark}
      style={{
        justifyContent: 'space-between',
      }}
    >
      <Text px={3} style={{ color: theme.colors.text.secondary }}>
        {userHost}
      </Text>

      <Flex px={3}>
        <Flex alignItems="center">
          <StyledFolderShared
            style={primaryOnTrue(isSharingDirectory)}
            pr={3}
            title={
              isSharingDirectory
                ? 'Directory Sharing Enabled'
                : 'Directory Sharing Disabled'
            }
          />
          <StyledClipboard
            style={primaryOnTrue(clipboardSharingEnabled)}
            pr={3}
            title={
              clipboardSharingEnabled
                ? 'Clipboard Sharing Enabled'
                : 'Clipboard Sharing Disabled'
            }
          />
          <StyledButton>ok?</StyledButton>
        </Flex>
        <ActionMenu
          onDisconnect={onDisconnect}
          showShareDirectory={canShareDirectory && !isSharingDirectory}
          onShareDirectory={onShareDirectory}
        />
      </Flex>
    </TopNav>
  );
}

export const TopBarHeight = 40;

const StyledClipboard = styled(Clipboard)`
  font-weight: ${({ theme }) => theme.fontWeights.bold};
  font-size: ${({ theme }) => theme.fontSizes[4] + 'px'};
  align-self: 'center';
`;

const StyledFolderShared = styled(FolderShared)`
  font-weight: ${({ theme }) => theme.fontWeights.bold};
  font-size: ${({ theme }) => theme.fontSizes[6] + 'px'};
  align-self: 'center';
`;

const StyledButton = styled(Button)`
  min-height: 0;
  height: ${({ theme }) => theme.fontSizes[7] + 'px'};
`;

type Props = {
  userHost: string;
  clipboardSharingEnabled: boolean;
  canShareDirectory: boolean;
  isSharingDirectory: boolean;
  onDisconnect: VoidFunction;
  onShareDirectory: VoidFunction;
};
