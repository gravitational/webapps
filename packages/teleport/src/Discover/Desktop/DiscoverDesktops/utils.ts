import { Path } from 'teleport/Discover/Desktop/DiscoverDesktops/lines/Path';
import { MonotoneX } from 'teleport/Discover/Desktop/DiscoverDesktops/lines/MonotoneX';

const STROKE_WIDTH = 4;

export interface Line {
  width: number;
  height: number;
  path: string;
}

export function createLine(
  desktopServiceElement: HTMLDivElement,
  desktopElement: HTMLDivElement,
  containerElement: HTMLDivElement
): Line {
  if (!desktopElement || !desktopServiceElement || !containerElement) {
    return null;
  }

  const desktopServiceRect = desktopServiceElement.getBoundingClientRect();
  const desktopRect = desktopElement.getBoundingClientRect();
  const containerRect = containerElement.getBoundingClientRect();

  const distance = desktopRect.left - desktopServiceRect.right;

  const path = new Path();
  const line = new MonotoneX(path);

  line.lineStart();

  const desktopLinePosition =
    desktopRect.top - containerRect.top + desktopRect.height / 2 - 1;
  const desktopServiceLinePosition =
    desktopServiceRect.top - containerRect.top + desktopServiceRect.height / 2;

  line.point(0, desktopServiceLinePosition - STROKE_WIDTH * 2);
  line.point(40, desktopServiceLinePosition - STROKE_WIDTH * 2);
  line.point(distance - 10, desktopLinePosition + STROKE_WIDTH / 2);
  line.point(distance, desktopLinePosition + STROKE_WIDTH / 2);

  line.lineEnd();

  return {
    width: distance,
    height: containerRect.height,
    path: path.path,
  };
}
