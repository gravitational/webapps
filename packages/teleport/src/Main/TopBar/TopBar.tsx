/*
Copyright 2019-2020 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import { NavLink } from 'react-router-dom';
import TopNavUserMenu from 'design/TopNav/TopNavUserMenu';
import { MenuItemIcon, MenuItem } from 'design/Menu';
import session from 'teleport/services/session';
import { useTeleport } from 'teleport/teleportContextProvider';
import { Flex, ButtonPrimary, TopNav } from 'design';
import TeleportContext from 'teleport/teleportContext';
import ClusterSelector from './ClusterSelector';
import { useTheme } from 'styled-components';
import history from 'teleport/services/history';
import cfg from 'teleport/config';

export default function Container() {
  const ctx = useTeleport();
  const state = useTopBar(ctx);
  return <TopBar {...state} />;
}

export function TopBar(props: ReturnType<typeof useTopBar>) {
  const {
    username,
    loadClusters,
    popupItems,
    changeCluster,
    clusterId,
  } = props;

  const [open, setOpen] = React.useState(false);
  const theme = useTheme();

  function showMenu() {
    setOpen(true);
  }

  function closeMenu() {
    setOpen(false);
  }

  function logout() {
    closeMenu();
    props.logout();
  }

  const menuItemProps = {
    onClick: closeMenu,
    py: 2,
    as: NavLink,
    exact: true,
  };

  const $userMenuItems = popupItems.map((item, index) => (
    <MenuItem {...menuItemProps} key={index} to={item.getLink(props.clusterId)}>
      <MenuItemIcon as={item.Icon} mr="2" />
      {item.title}
    </MenuItem>
  ));

  return (
    <TopNav
      height="56px"
      bg="inherit"
      pl="5"
      style={{
        overflowY: 'initial',
        flexShrink: '0',
        borderBottom: `1px solid ${theme.colors.primary.main}`,
      }}
    >
      <ClusterSelector
        value={clusterId}
        width="400px"
        maxMenuHeight={200}
        mr="20px"
        onChange={changeCluster}
        onLoad={loadClusters}
      />
      <Flex ml="auto" height="100%">
        <TopNavUserMenu
          menuListCss={menuListCss}
          open={open}
          onShow={showMenu}
          onClose={closeMenu}
          user={username}
        >
          {$userMenuItems}
          <MenuItem>
            <ButtonPrimary my={3} block onClick={logout}>
              Sign Out
            </ButtonPrimary>
          </MenuItem>
        </TopNavUserMenu>
      </Flex>
    </TopNav>
  );
}

const menuListCss = () => `
  width: 250px;
`;

function useTopBar(ctx: TeleportContext) {
  const clusterId = ctx.storeClusterId.getId();
  const popupItems = ctx.storeNav.getTopMenuItems();
  const { username } = ctx.storeUser.state;

  function loadClusters() {
    return ctx.clusterService.fetchClusters();
  }

  function logout() {
    session.logout();
  }

  function changeCluster(value: string) {
    const loc = history.getLocation() as Location;
    const newPrefix = cfg.getClusterRoute(value);
    const oldPrefix = cfg.getClusterRoute(clusterId);
    const newPath = loc.pathname.replace(oldPrefix, newPrefix);
    history.push(newPath);
  }

  return {
    popupItems,
    username,
    clusterId,
    changeCluster,
    loadClusters,
    logout,
  };
}
