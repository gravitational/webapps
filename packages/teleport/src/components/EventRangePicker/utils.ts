import moment from 'moment';

export function getRangeOptions(): EventRange[] {
  return [
    {
      name: 'Today',
      from: moment(new Date())
        .startOf('day')
        .toDate(),
      to: moment(new Date())
        .endOf('day')
        .toDate(),
    },
    {
      name: '7 days',
      from: moment()
        .subtract(6, 'day')
        .startOf('day')
        .toDate(),
      to: moment(new Date())
        .endOf('day')
        .toDate(),
    },
    {
      name: 'Custom Range...',
      isCustom: true,
      from: new Date(),
      to: new Date(),
    },
  ];
}

export type EventRange = {
  from: Date;
  to: Date;
  isCustom?: boolean;
  name?: string;
};
