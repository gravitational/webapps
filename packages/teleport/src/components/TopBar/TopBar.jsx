/*
Copyright 2019-2020 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import TopNavUserMenu from 'design/TopNav/TopNavUserMenu';
import { MenuItemIcon, MenuItem } from 'design/Menu';
import teleportLogoSvg from 'design/assets/images/teleport-logo.svg';
import { withState } from 'shared/hooks';
import session from 'teleport/services/session';
import { useStoreUser, useStoreNav } from 'teleport/teleportContextProvider';
import { Image, Flex, ButtonPrimary, TopNav, TopNavItem } from 'design';
import cfg from 'teleport/config';

export class DashboardTopNav extends React.Component {
  state = {
    open: false,
  };

  onShowMenu = () => {
    this.setState({ open: true });
  };

  onCloseMenu = () => {
    this.setState({ open: false });
  };

  onItemClick = () => {
    this.onClose();
  };

  onLogout = () => {
    this.onCloseMenu();
    this.props.onLogout();
  };

  menuItemProps = {
    onClick: this.onCloseMenu,
    py: 2,
    as: NavLink,
    exact: true,
  };

  render() {
    const { username, topMenuItems, pl, children } = this.props;
    const { open } = this.state;
    const $userMenuItems = topMenuItems.map((item, index) => (
      <MenuItem {...this.menuItemProps} key={index} to={item.to}>
        <MenuItemIcon as={item.Icon} mr="2" />
        {item.title}
      </MenuItem>
    ));

    return (
      <TopNav
        height="56px"
        pl={pl}
        style={{
          zIndex: '1',
          boxShadow: '0 4px 16px rgba(0,0,0,.24)',
          overflowX: 'auto',
        }}
      >
        <TopNavItem width="208px" as={Link} to={cfg.routes.app}>
          <Image
            src={teleportLogoSvg}
            mx="3"
            maxHeight="24px"
            maxWidth="160px"
          />
        </TopNavItem>
        {children}
        <Flex ml="auto" height="100%">
          <TopNavUserMenu
            menuListCss={menuListCss}
            open={open}
            onShow={this.onShowMenu}
            onClose={this.onCloseMenu}
            user={username}
          >
            {$userMenuItems}
            <MenuItem>
              <ButtonPrimary my={3} block onClick={this.onLogout}>
                Sign Out
              </ButtonPrimary>
            </MenuItem>
          </TopNavUserMenu>
        </Flex>
      </TopNav>
    );
  }
}

const menuListCss = () => `
  width: 250px;
`;

function mapState() {
  const topMenuItems = useStoreNav().getTopMenuItems();
  const { username } = useStoreUser().state;
  return {
    topMenuItems,
    username,
    onLogout: () => session.logout(),
  };
}

export default withState(mapState)(DashboardTopNav);
