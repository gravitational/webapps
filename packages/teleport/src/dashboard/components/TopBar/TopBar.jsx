import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import TopNavUserMenu from 'design/TopNav/TopNavUserMenu';
import { MenuItem } from 'design/Menu';
import teleportLogoSvg from 'design/assets/images/teleport-logo.svg';
import { withState } from 'shared/hooks';
import session from 'teleport/services/session';
import { useStoreUser, useStoreNav } from 'teleport/teleportContextProvider';
import { Image, Flex, ButtonPrimary, TopNav, TopNavItem } from 'design';
import cfg from 'teleport/config';

export class DashboardTopNav extends React.Component {
  state = {
    open: false,
  };

  onShowMenu = () => {
    this.setState({ open: true });
  };

  onCloseMenu = () => {
    this.setState({ open: false });
  };

  onItemClick = () => {
    this.onClose();
  };

  onLogout = () => {
    this.onCloseMenu();
    this.props.onLogout();
  };

  menuItemProps = {
    onClick: this.onCloseMenu,
    py: 2,
    as: NavLink,
    exact: true,
  };

  render() {
    const { username, items = [], pl } = this.props;
    const { open } = this.state;

    const $items = items.map((item, index) => (
      <TopNavItem
        px="5"
        as={NavLink}
        exact={item.exact}
        key={index}
        to={item.to}
        style={{
          textTransform: 'uppercase',
        }}
      >
        {item.title}
      </TopNavItem>
    ));

    return (
      <TopNav
        height="56px"
        pl={pl}
        style={{ zIndex: '1', boxShadow: '0 8px 24px rgba(0,0,0,.24)' }}
      >
        <TopNavItem pr="10" as={Link} to={cfg.routes.app}>
          <Image
            src={teleportLogoSvg}
            mx="3"
            maxHeight="24px"
            maxWidth="160px"
          />
        </TopNavItem>
        {$items}
        <Flex ml="auto" height="100%">
          <TopNavUserMenu
            menuListCss={menuListCss}
            open={open}
            onShow={this.onShowMenu}
            onClose={this.onCloseMenu}
            user={username}
          >
            <MenuItem>
              <ButtonPrimary my={3} block onClick={this.onLogout}>
                Sign Out
              </ButtonPrimary>
            </MenuItem>
          </TopNavUserMenu>
        </Flex>
      </TopNav>
    );
  }
}

const menuListCss = () => `
  width: 250px;
`;

function mapState() {
  const items = useStoreNav().getTopItems();
  const { username } = useStoreUser().state;
  return {
    items,
    username,
    onLogout: () => session.logout(),
  };
}

export default withState(mapState)(DashboardTopNav);
