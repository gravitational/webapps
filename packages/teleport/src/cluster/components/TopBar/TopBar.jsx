/*
Copyright 2019 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import styled from 'styled-components';
import { ArrowRight } from 'design/Icon';
import { NavLink } from 'react-router-dom';
import { Text, ButtonOutlined, Flex } from 'design';
import TopBar from 'teleport/components/TopBar';
import cfg from 'teleport/config';

export default function ClusterTopBar({ onClickClusterInfo }) {
  const clusterId = cfg.clusterName;
  return (
    <TopBar>
      <Flex alignItems="center">
        <BreadCrumbLink
          ml="6"
          typography="h5"
          color="text.secondary"
          as={NavLink}
          to={cfg.routes.app}
          title="Go back to cluster list"
        >
          All Clusters
        </BreadCrumbLink>
        <ArrowRight mx="2" fontSize={12} color="text.secondary" />
        <Text
          mr="3"
          typography="h5"
          color="text.primary"
          style={{ maxWidth: '200px', whiteSpace: 'nowrap' }}
          title={clusterId}
        >
          {clusterId}
        </Text>
        <ButtonOutlined
          size="small"
          style={{ whiteSpace: 'nowrap' }}
          onClick={onClickClusterInfo}
        >
          Cluster Info
        </ButtonOutlined>
      </Flex>
    </TopBar>
  );
}

const BreadCrumbLink = styled(Text)`
  text-decoration: none;
  font-size: 14px;
  font-weight: 300;
  white-space: nowrap;

  &:hover {
    color: ${props => props.theme.colors.text.primary};
  }
`;
