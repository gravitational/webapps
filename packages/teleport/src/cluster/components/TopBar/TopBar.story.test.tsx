import React from 'react';
import { ClusterTopBar } from './TopBar.story';
import { render } from 'design/utils/testing';

test('rendering of ClusterTopBar', () => {
  const { container } = render(<ClusterTopBar />);
  expect(container.firstChild).toMatchSnapshot();
});
