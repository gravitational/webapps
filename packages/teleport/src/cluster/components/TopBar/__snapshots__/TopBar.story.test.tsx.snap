// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of ClusterTopBar 1`] = `
.c5 {
  display: inline-block;
  transition: color .3s;
  margin-left: 8px;
  margin-right: 8px;
  color: rgba(255,255,255,0.56);
  font-size: 12px;
}

.c7 {
  line-height: 1.5;
  border-radius: 4px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  border: 1px solid;
  box-sizing: border-box;
  background-color: transparent;
  cursor: pointer;
  font-family: inherit;
  font-weight: bold;
  outline: none;
  opacity: .56;
  position: relative;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  transition: all .3s;
  -webkit-font-smoothing: antialiased;
  border-color: rgba(255,255,255,0.87);
  color: #FFFFFF;
  font-size: 10px;
  min-height: 24px;
  padding: 0px 16px;
  border-color: rgba(255,255,255,0.87);
  color: rgba(255,255,255,0.87);
}

.c7:hover {
  opacity: 1;
}

.c7:active {
  opacity: .24;
}

.c7 > span {
  display: flex;
  align-items: center;
  justify-content: center;
}

.c7:hover,
.c7:focus {
  border-color: #FFFFFF;
  color: #FFFFFF;
}

.c7:disabled {
  background: rgba(255,255,255,0.12);
  color: rgba(255,255,255,0.3);
}

.c7:hover,
.c7:focus {
  border-color: #FFFFFF;
  color: #FFFFFF;
}

.c2 {
  display: block;
  outline: none;
  margin-left: 16px;
  margin-right: 16px;
  max-width: 160px;
  max-height: 24px;
}

.c6 {
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  margin: 0px;
  margin-right: 16px;
  color: rgba(255,255,255,0.87);
}

.c10 {
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 600;
  font-size: 12px;
  margin: 0px;
}

.c0 {
  box-sizing: border-box;
  height: 56px;
  background-color: #1C254D;
  flex: 0 0 auto;
  display: flex;
  align-items: center;
  flex-direction: row;
}

.c3 {
  box-sizing: border-box;
  display: flex;
  align-items: center;
}

.c8 {
  box-sizing: border-box;
  margin-left: auto;
  height: 100%;
  display: flex;
}

.c1 {
  align-items: center;
  background: none;
  border: none;
  color: rgba(255,255,255,.56);
  cursor: pointer;
  display: inline-flex;
  font-size: 11px;
  font-weight: 600;
  height: 100%;
  margin: 0;
  outline: none;
  padding: 0 16px;
  position: relative;
  text-decoration: none;
  width: 208px;
}

.c1:hover,
.c1:focus {
  background: rgba(255,255,255,.06);
}

.c1.active {
  background: #222C59;
  color: #FFFFFF;
}

.c1.active:after {
  background-color: #651FFF;
  content: "";
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 4px;
}

.c9 {
  align-items: center;
  background: none;
  border: none;
  color: rgba(255,255,255,.56);
  cursor: pointer;
  display: inline-flex;
  font-size: 11px;
  font-weight: 600;
  height: 100%;
  margin: 0;
  outline: none;
  padding: 0 16px;
  position: relative;
  text-decoration: none;
  margin-left: auto;
  max-width: 250px;
}

.c9:hover,
.c9:focus {
  background: rgba(255,255,255,.06);
}

.c9.active {
  background: #222C59;
  color: #FFFFFF;
}

.c9.active:after {
  background-color: #651FFF;
  content: "";
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 4px;
}

.c11 {
  align-items: center;
  background: #651FFF;
  border-radius: 50%;
  display: flex;
  font-size: 14px;
  font-weight: bold;
  justify-content: center;
  height: 32px;
  margin-left: 16px;
  width: 100%;
  max-width: 32px;
  min-width: 32px;
}

.c4 {
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  margin: 0px;
  margin-left: 40px;
  color: rgba(255,255,255,0.56);
  text-decoration: none;
  font-size: 14px;
  font-weight: 300;
  white-space: nowrap;
}

.c4:hover {
  color: rgba(255,255,255,0.87);
}

<nav
  class="sc-AxiKw c0"
  height="56px"
  style="z-index: 1; box-shadow: 0 4px 16px rgba(0,0,0,.24); overflow-x: auto;"
>
  <a
    class="c1"
    href="/web"
    width="208px"
  >
    <img
      class="c2"
      src="file_stub"
    />
  </a>
  <div
    class="sc-AxiKw c3"
  >
    <a
      class="c4"
      color="text.secondary"
      href="/web"
      m="0"
      ml="6"
      theme="[object Object]"
      title="Go back to cluster list"
      typography="h5"
    >
      All Clusters
    </a>
    <span
      class="c5 icon icon-chevron-right "
      color="text.secondary"
      font-size="12"
    />
    <div
      class="c6"
      color="text.primary"
      style="max-width: 200px; white-space: nowrap;"
      title="localhost"
    >
      localhost
    </div>
    <button
      class="c7"
      style="white-space: nowrap;"
    >
      <span>
        Cluster Info
      </span>
    </button>
  </div>
  <div
    class="sc-AxiKw c8"
    height="100%"
  >
    <button
      class="c9"
    >
      <div
        class="c10"
        font-size="12px"
      >
        Timothy Kim
      </div>
      <div
        class="c11"
      >
        T
      </div>
    </button>
  </div>
</nav>
`;
